### PR TITLE
make it compile

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: src/kpro.erl [hunk 1]
```diff
@@ -84,497 +98,587 @@
...
+%% @hidden Get predefined schema from kpro_schema:get/2.
+-spec get_schema(req_tag() | rsp_tag(), vsn()) -> struct_schema().
+get_schema(Tag, Vsn) ->
+  get_schema(?SCHEMA_MODULE, Tag, Vsn).
+
+%% @hidden Get predefined schema from Module:get/2 API.
+-spec get_schema(module(), req_tag() | rsp_tag(), vsn()) -> struct_schema().
+get_schema(Module, Tag, Vsn) ->
+  try
+    Module:get(Tag, Vsn)
```
Comment (id) [inline src/kpro.erl:397]: why do we need custom callbacks?

File: src/kpro.erl [hunk 2]
```diff
@@ -84,497 +98,587 @@
...
-        decode_message_stream(java_snappy_unpack(Msg#kpro_message.value), Acc);
-      #kpro_message{attributes = Attr} = Msg when ?KPRO_IS_LZ4_ATTR(Attr) ->
-        decode_message_stream(lz4_unpack(Msg#kpro_message.value), Acc);
-      _Else ->
-        [Msg | Acc]
-    end,
-  decode_message_stream(Rest, NewAcc).
+%% @private Encode embedded bytes.
+-spec enc_embeded(stack(), field_value()) -> field_value().
+enc_embeded(_Stack, Value) -> Value.
```
Comment (id) [inline src/kpro.erl]: embedded

File: src/kpro.erl [hunk 3]
```diff
@@ -84,497 +98,587 @@
...
+                | {array, struct_schema()}.
+-type stack() :: [term()]. %% encode / decode stack
 
 -define(INT, signed-integer).
+-define(SCHEMA_MODULE, kpro_schema).
+
+%%%_* APIs =====================================================================
 
-%% @doc Help function to contruct a #kpro_offsets_request_v0{} for requests
+%% @doc Help function to contruct a OffsetsRequest
```
Comment (id) [inline src/kpro.erl:140]: so the plan is to have methods for all of requests?

File: src/kpro.erl (hunk 1 above)
Comment (zmstone) [reply src/kpro.erl:397]: there is also `kpro_prelude_schema`

**Review by id — APPROVED** [review]

**Pull request merged**